navit/start_real.c

```diff
@@ -51,6 +51,10 @@
 #include <winbase.h>
 #endif
 
+#if IOS
+#include "gthreadprivate.h"
+#endif
```

Comment:
If not available, it produces an error with SDK 14.5. With SDK 10 it is only a warning. There is a missing prototype.

Reply:
i am sure this would fix the issue, but its not the best fix possible. could you post the full error text (gist or pastebin)? Probably the include is missing within support/glib and can be added for every System/Build/Port

Reply:
CompileC build/navit/navit.build/Debug-iphoneos/navit_core.build/Objects-normal/arm64/start_real.o navit/start_real.c normal arm64 c com.apple.compilers.llvm.clang.1_0.compiler
    cd /Volumes/Data/Users/olf/navitios/navit
    export LANG=en_US.US-ASCII
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c -target arm64-apple-ios14.5 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -DCMAKE_INTDIR=\"Debug-iphoneos\" -DMODULE=navit -DLIBDIR=\"/usr/local/lib/navit\" -DPREFIX=\"/usr/local\" -DBIN_DIR=\"bin\" -DSHARE_DIR=\"share/navit\" -DLOCALE_DIR=\"share/locale\" -DIMAGE_DIR=\"share/navit/icons\" -DTEXTURE_DIR=\"share/navit/textures\" -DMAN_DIR=\"share/man/man1\" -DLIB_DIR=\"lib/navit\" -DIOS -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.5.sdk -fstrict-aliasing -Wdeprecated-declarations -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-semicolon-before-method-body -fembed-bitcode-marker -index-store-path /Volumes/Data/Users/olf/Library/Developer/Xcode/DerivedData/navit-ghxqlrmlampycecusralpmsjrtkj/Index/DataStore -I/Volumes/Data/Users/olf/navitios/navit/build/navit/Debug-iphoneos/include -I/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS14.5.sdk/usr/include -I/Volumes/Data/Users/olf/navitios/navit/build -I/Volumes/Data/Users/olf/navitios/navit -I/Volumes/Data/Users/olf/navitios/navit/navit -I/Volumes/Data/Users/olf/navitios/navit/build/navit -I/Volumes/Data/Users/olf/navitios/navit/navit/support -I/Volumes/Data/Users/olf/navitios/navit/navit/support/ezxml -I/Volumes/Data/Users/olf/navitios/navit/navit/support/glib -I/Volumes/Data/Users/olf/navitios/navit/navit/support/libpng -I/Volumes/Data/Users/olf/navitios/navit/navit/support/wordexp -I/Volumes/Data/Users/olf/navitios/navit/navit/support/gettext_intl -I/Volumes/Data/Users/olf/navitios/navit/navit/graphics/null -I/Volumes/Data/Users/olf/navitios/navit/navit/osd/core -I/Volumes/Data/Users/olf/navitios/navit/navit/vehicle/demo -I/Volumes/Data/Users/olf/navitios/navit/navit/vehicle/file -I/Volumes/Data/Users/olf/navitios/navit/navit/gui/internal -I/Volumes/Data/Users/olf/navitios/navit/navit/map/binfile -I/Volumes/Data/Users/olf/navitios/navit/navit/map/filter -I/Volumes/Data/Users/olf/navitios/navit/navit/map/mg -I/Volumes/Data/Users/olf/navitios/navit/navit/map/shapefile -I/Volumes/Data/Users/olf/navitios/navit/navit/map/textfile -I/Volumes/Data/Users/olf/navitios/navit/navit/map/csv -I/Volumes/Data/Users/olf/navitios/navit/navit/speech/iphone -I/Volumes/Data/Users/olf/navitios/navit/navit/traffic/dummy -I/Volumes/Data/Users/olf/navitios/navit/navit/traffic/null -I/Volumes/Data/Users/olf/navitios/navit/navit/vehicle/iphone -I/Volumes/Data/Users/olf/navitios/navit/navit/graphics/cocoa -I/Volumes/Data/Users/olf/navitios/navit/navit/fib-1.1 -I/Volumes/Data/Users/olf/navitios/navit/build/navit/navit.build/Debug-iphoneos/navit_core.build/DerivedSources-normal/arm64 -I/Volumes/Data/Users/olf/navitios/navit/build/navit/navit.build/Debug-iphoneos/navit_core.build/DerivedSources/arm64 -I/Volumes/Data/Users/olf/navitios/navit/build/navit/navit.build/Debug-iphoneos/navit_core.build/DerivedSources -F/Volumes/Data/Users/olf/navitios/navit/build/navit/Debug-iphoneos -fembed-bitcode -MMD -MT dependencies -MF /Volumes/Data/Users/olf/navitios/navit/build/navit/navit.build/Debug-iphoneos/navit_core.build/Objects-normal/arm64/start_real.d --serialize-diagnostics /Volumes/Data/Users/olf/navitios/navit/build/navit/navit.build/Debug-iphoneos/navit_core.build/Objects-normal/arm64/start_real.dia -c /Volumes/Data/Users/olf/navitios/navit/navit/start_real.c -o /Volumes/Data/Users/olf/navitios/navit/build/navit/navit.build/Debug-iphoneos/navit_core.build/Objects-normal/arm64/start_real.o

/Volumes/Data/Users/olf/navitios/navit/navit/start_real.c:90:5: error: implicit declaration of function '_g_slice_thread_init_nomessage' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    _g_slice_thread_init_nomessage();
    ^
1 error generated.

Reply:
The method is in support_glib/gslice.c which includes  "gthreadprivate.h" but not "gslice.h". 
If removing the import of gthreadprivate.h from gslice.c, adding an include of gslice.h and adding the include for gthreadprivate.h in gslice.h it works. That seems to be a solution.

Reply:
Changes applied, CI for all builds OK.